src/main/software/amazon/event/ruler/Configuration.java

```diff
@@ -0,0 +1,42 @@
+package software.amazon.event.ruler;
+
+/**
+ * Configuration for a Machine.
+ */
+public class Configuration {
+
+    /**
+     * Normally, NameStates are re-used for a given key subsequence and pattern if this key subsequence and pattern have
```

Comment:
Just to note that this would need quite a bit of work before it went in the README.  I found it really hard to understand. I suggest that the easiest way to make it comprehensible to mere mortals is to include a few examples.

Reply:
Not sure it's "quite a bit of work", but I added to the readme with an example.